contracts/protocol/v1/proxies/P1CurrencyConverterProxy.sol

```diff
@@ -0,0 +1,236 @@
+/*
+
+    Copyright 2020 dYdX Trading Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+*/
+
+pragma solidity 0.5.16;
+pragma experimental ABIEncoderV2;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import { I_ExchangeWrapper } from "../../../external/I_ExchangeWrapper.sol";
+import { I_PerpetualV1 } from "../intf/I_PerpetualV1.sol";
+
+
+/**
+ * @title P1CurrencyConverterProxy
+ * @author dYdX
+ *
+ * @notice Proxy contract which executes a trade via an ExchangeWrapper before making a deposit
+ *  or after making a withdrawal.
+ */
+contract P1CurrencyConverterProxy {
+    using SafeERC20 for IERC20;
+
+    // ============ Events ============
+
+    event LogConvertedDeposit(
```

Comment:
nit: add `address source` or `address origin` to have "parallel" fields to `LogConvertedWithdrawal`?

Reply:
Good idea, I think I prefer `source`